### ISSUE TITLE
Assistants V2 breaks assistant creation

### ISSUE THREAD

**Issue by phact** [phact/streaming-assistants#2]
Since openai assistants v2 has breaking changes, the `OpenAI-Beta: assistants=v1` header is required when using recent versions of the sdk

```
Traceback (most recent call last):
  File "/python/helpers/pydev/pydevd.py", line 1534, in _exec
    pydev_imports.execfile(file, globals, locals)  # execute the script
  File "/pydev/_pydev_imps/_pydev_execfile.py", line 18, in execfile
    exec(compile(contents+"\n", file, 'exec'), glob, loc)
  File "./streaming-assistants/examples/run_retreival.py", line 82, in <module>
    run_with_assistant(gpt3_assistant, client)
  File "./streaming-assistants/examples/run_retreival.py", line 45, in run_with_assistant
    assistant = client.beta.assistants.update(
  File "./streaming-assistants/streaming_assistants/patch.py", line 506, in patched_method
    result = original_method(self, *args, **kwargs)
TypeError: Assistants.update() got an unexpected keyword argument 'file_ids'
```

**Comment by phact** [comment]
To make this simpler I released [`v0.16.0`](https://pypi.org/manage/project/streaming-assistants/release/0.16.0/) which pins dependencies to supported openai sdk versions until we add v2 support in https://github.com/datastax/astra-assistants-api/issues/22

**Comment by phact** [comment]
Server now disambiguates between header versions and v2 support is next.
